frontend/scripts/build-vanilla-extract.mjs

```diff
@@ -1,103 +1,131 @@
-		absWorkingDir: rootDirectory,
-		plugins: [
-			vanillaExtractPlugin({
-				// We don't output css here, only the js modules for each stylesheet entrypoint
-				outputCss: false,
-				identifiers: 'short',
-			}),
-		],
-		external: [],
-		logLevel: 'warning',
-	})
-	console.log(new Date(), 'built vanilla extract entry points')
-	return
-}
+	const packageJson = JSON.parse(
+		fs.readFileSync(path_.join(rootDirectory, 'package.json')),
+	)
 
-const isEntryPoint = (path) => path.endsWith('.css.ts')
+	// uncomment for cleanup
+	// await fs.promises.rm(outputDirectory, { recursive: true, force: true })
 
-if (watch) {
-	// No await here since we don't want to delay watcher and miss updates
-	build()
+	let entryPoints = Object.fromEntries(
+		(
+			await readdirp.promise(workingDirectory, {
+				fileFilter: entryPointGlobs,
+				directoryFilter: ['!__generated', '!graph/generated'],
+			})
+		).map(({ path }) => [path, path_.join(workingDirectory, path)]),
+	)
 
-	chokidar
-		.watch(inputGlobs, {
-			cwd: workingDirectory,
-			ignoreInitial: true,
-			ignored: [
-				'**/__generated/**/**',
-				'**/graph/generated/**/**',
-				'**/**.d.ts',
+	const build = async () => {
+		console.log(new Date(), 'building vanilla extract entry points')
+		await esbuild.build({
+			entryPoints: Object.values(entryPoints),
+			bundle: true,
+			// TODO: support user supplied sourcemaps
+			// sourcemap: true,
+			sourcemap: false,
+			format: 'esm',
+			platform: 'browser',
+			outdir: outputDirectory,
+			outbase: workingDirectory,
+			minify: true,
+			splitting: true,
+			target: 'esnext',
+			// Working directory apparently affects vanilla extract's hashes?
+			// Using rootDirectory to stay consistent with vite setup
+			absWorkingDir: rootDirectory,
+			plugins: [
+				vanillaExtractPlugin({
+					// We don't output css here, only the js modules for each stylesheet entrypoint
+					outputCss: false,
+					identifiers: 'short',
+				}),
 			],
+			external: [
+				...Object.keys(packageJson.dependencies).flatMap((pkg) => [
+					pkg,
+					`${pkg}/*`,
+				]),
+			],
+			logLevel: 'warning',
 		})
-		.on('add', (path) => {
-			if (isEntryPoint(path)) {
-				console.log(
-					new Date(),
-					'adding vanilla extract entry point',
-					path,
-				)
-				entryPoints[path] = path_.join(workingDirectory, path)
-			}
-			// We'll probably run into race conditions with concurrent edits here...
-			// TODO: make it more robust with some debouncing + transaction tracking
-			build(path)
-		})
-		.on('change', (path) => build(path))
-		.on('unlink', (path) => {
-			if (isEntryPoint(path)) {
-				delete entryPoints[path]
-			}
-			build(path)
-		})
-		.on('ready', () => console.log(new Date(), 'ready'))
-} else {
-	await build()
+		console.log(new Date(), 'built vanilla extract entry points')
+		return
+	}
+
+	const isEntryPoint = (path) => path.endsWith('.css.ts')
+
+	if (watch) {
+		let transactionIdLatest = randomString()
```

Comment:
This removes some race conditions and ensures at most 1 instance of build is running at a time, which helps put an upper bound on memory usage as well.

Reply:
neat!